### PR TITLE
About Us

### PR DESCRIPTION
**Pull request by AdrianJakubcik**
The parser is far from finished, but it gets its job done atm. I'm planning on adding functions to save objects into a json file so that it is easier to edit them, another advantage of them being in a JSON file is the ability and possibility to add pictures to individual cards, and much more...

### CODE REVIEW

File: Extensions/AboutUsParser.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+async function createFilesArray(files) {
+    let mappedFiles = [];
+    files.map(file => {
+        file = file.split('/');
+        mappedFiles.push({ name: file[file.length - 1], path: Path.join(...file), language: file[file.length - 2], folder: file[file.length - 3] });
+    });
+    return mappedFiles;
+}
+
+/** Parses given string by separators and empty lines
```
Comment (DiscreteConcrete) [inline Extensions/AboutUsParser.js:93]: parses into what?

File: Extensions/AboutUsParser.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+        sections.map(section => { // We iterate through every parsed section
+            let para = section.split('\r\n').filter(sentence => sentence.trim() ? true : false); // Splitting sections into paragraphs by new line and removing empty lines
+            //console.log(para.slice(2, para.length));
+            parsedSections.push({ title: para[0], short: para[1], long: para.slice(2,para.length) }); // Pushing a parsed section object into an array
+        });
+
+        return parsedSections; // return array of parsed sections as a callback
+    } catch (error) {
+        console.error(error);
```
Comment (DiscreteConcrete) [inline Extensions/AboutUsParser.js:112]: why are we catching the errors? shouldn't they break the script instead?

File: Extensions/AboutUsParser.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    }
+}
+
+/**
+ * 
+ * @param {Array} array 
+ */
+function getStringFromArray(array) {
+    let str_builder = '';
+    //console.log(array);
```
Comment (DiscreteConcrete) [inline Extensions/AboutUsParser.js:122]: ?

File: Extensions/AboutUsParser.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    } catch (error) {
+        console.error(error);
+    }
+}
+
+/**
+ * 
+ * @param {Array} array 
+ */
+function getStringFromArray(array) {
```
Comment (DiscreteConcrete) [inline Extensions/AboutUsParser.js:120]: what's this for?

File: Extensions/AboutUsParser.js [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+}
+
+/**
+ * @param {String} Fpath Path where to create the component
+ * @param {String} name File name of the component
+ * @param {String} string String that will be written inside the component
+ * @param {String} language
+ * @param {Boolean} append_file Whether to append or overwrite already existing file 
+ */
+async function createComponent(Fpath = null, name, string, language, append_file = true, fileName = 'school') {
```
Comment (DiscreteConcrete) [inline Extensions/AboutUsParser.js:136]: what's this for?

Comment (DiscreteConcrete) [inline Extensions/AboutUsParser.js:136]: please let's not use abbreviations where not needed. this is so much harder to read

Comment (DiscreteConcrete) [reply Extensions/AboutUsParser.js:136]: and why is it with a capital?

File: Extensions/AboutUsParser.js [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+
+/** Creates a Json file with our data
+ * 
+ * @param {Array} array Array of paragraph objects
+ */
+function saveAsJson(array) {
+    fs.writeFileSync(C.dPath + C.jsonFile, JSON.stringify(array, null, 4), {flag: 'w+', mode: 0666});
+}
+
+/** Loads a Json file and returns its content decoded
```
Comment (DiscreteConcrete) [inline Extensions/AboutUsParser.js:181]: decoded? you mean parsed?

File: Extensions/AboutUsParser.js [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+ */
+function checkJsonForPhotos(parsedJson) {
+    let arr = parsedJson;
+
+    return parsedJson;
+}
+
+Files(C.dPath, async res => {
+    var arr = await createFilesArray(res);
+    var folders = getFolders(C.dPath);
```
Comment (DiscreteConcrete) [inline Extensions/AboutUsParser.js:220]: why are some thing `var` and some `let`?

File: Extensions/AboutUsParser.js [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+ * @param {Array} array 
+ * @deprecated This will be added later on, it should check whether Json file contains objects with paths to photos and ask whether to combine them with newly parsed objects array of data
+ */
+function checkJsonForPhotos(parsedJson) {
+    let arr = parsedJson;
+
+    return parsedJson;
+}
+
+Files(C.dPath, async res => {
```
[the remark this answers is not in the document]
Comment (DiscreteConcrete) [reply Extensions/AboutUsParser.js:218]: can you add overall description and also within the block?

File: components/navigation.html [hunk 9]
```diff
@@ -8,13 +8,20 @@
         <div class="left">
             <!-- Search button -->
             <div class="search-container close">
-                <input type="search" name="searchInp" onchange="sendSearch(this.value, 'search.html')" onkeyup="searchSitesWithResult(this.value, 'search_results', false, true),searchInSite(this.value)" id="search_box" />
+                <!-- <input type="search" name="searchInp" id="searchInp" onchange="sendSearch(this.value, 'search.html')" onkeyup="searchSitesWithResult(this.value, 'search_results', false, true),searchInSite(this.value)" id="search_box" /> -->
```
Comment (DiscreteConcrete) [inline components/navigation.html:11]: why is this commented out?

File: components/summerSchool.html [hunk 10]
```diff
@@ -0,0 +1,28 @@
+:^) Cards :::
+<!-- Card 1 -->
+<div id="1" class="card admiss" onclick="Click(this)">
+    <h4 class="card-title">{{summer.primarySchool-title}}</h4>
+    <img src="{fill_parents}images/card_thumbnails/primary_school.jpg" class="card-img-top" alt="Picture Not Available!">
+    <div class="no-display admiss-text">
+        <object data="{fill_parents}files/summerschool/{{summer.primarySchool-file}}" type="image/png" width="100%" height="100%" frameborder="0">
+            <!-- <div class="alt-text">
+                <p>Previewing or manipulating a pdf file is not supported on your device's browser. We suggest you download the file and open it manually with a pdf reader.</p>
```
Comment (DiscreteConcrete) [inline components/summerSchool.html:9]: why is this commented out / still here?

File: components/summerSchool.html [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+</div>
+
+<!-- Card 2 -->
+<div id="2" class="card admiss" onclick="Click(this)">
+    <h4 class="card-title">{{summer.kindergarten-title}}</h4>
+    <img src="{fill_parents}images/card_thumbnails/kindergarten.jpg" class="card-img-top" alt="Picture Not Available!">
+    <div class="no-display admiss-text">
+        <object data="{fill_parents}files/summerschool/{{summer.kindergarten-file}}" type="image/png" width="100%" height="100%" frameborder="0">
+            <!-- <div class="alt-text">
+                <p>Previewing or manipulating a pdf file is not supported on your device's browser. We suggest you download the file and open it manually with a pdf reader.</p>
```
Comment (DiscreteConcrete) [inline components/summerSchool.html:23]: same

File: source/css/aboutus.css [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    .card.about .card-content {
+        width: calc(55%*0.99);
+        margin: 0 auto;
+        position: relative;
+        display: flex;
+        flex-direction: column;
+        justify-content: space-between;
+        /* height: max-content; */
```
Comment (DiscreteConcrete) [inline source/css/aboutus.css:114]: why is this here / commented out?

File: source/css/admissions.css [hunk 13]
```diff
@@ -1,12 +1,40 @@
+/* Second option
```
Comment (DiscreteConcrete) [inline source/css/admissions.css:1]: what does this mean? what second option?

File: source/css/admissions.css [hunk 14]
```diff
@@ -1,12 +1,40 @@
...
+} */
+
+#interviews-heading.admissions {
+    display: block !important;
+    font-size: xxx-large;
+    padding: 8px;
+    margin: 18px 0;
+    color: #a60000;
+    text-transform: capitalize;
+    /* text-decoration: underline; */
```
Comment (DiscreteConcrete) [inline source/css/admissions.css:34]: ?

File: source/css/nav_style.css [hunk 15]
```diff
@@ -688,91 +799,129 @@ span.searcher-result-preview:last-child {
...
         cursor: pointer;
         transition: ease 0.7s;
     }
+
 }
 
+
 @media screen and (max-width: 299px) {
-    .nav-phone .title {
+    /* .nav-phone .title {
```
Comment (DiscreteConcrete) [inline source/css/nav_style.css:864]: ?

File: source/js/about.js [hunk 16]
```diff
@@ -0,0 +1,36 @@
+$(document).ready(function() {
+
+    $('.toggle-btn').click(toggleShowLongText);
+
+    function toggleShowLongText() {
+        let caller = this.parentElement.parentElement;
+        // let skip_content_styling = caller.getAttribute('style') ? false : true;
```
Comment (DiscreteConcrete) [inline source/js/about.js:7]: ?

File: source/js/about.js [hunk 17]
```diff
@@ -0,0 +1,36 @@
+$(document).ready(function() {
+
+    $('.toggle-btn').click(toggleShowLongText);
+
+    function toggleShowLongText() {
```
Comment (DiscreteConcrete) [inline source/js/about.js:5]: ?

File: source/js/about.js [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+        let textContainer = caller.children[1].children[0];
+        textContainer.id = textContainer.getAttribute('id') ? textContainer.id : 'this';
+
+        let descripContainer = caller.children[0].children[1].children[0];
+        descripContainer.classList = descripContainer.getAttribute('class') ? '' : 'full';
+
+        $("#" + textContainer.id).slideToggle(250, () => {
+            if (img && !skip_img_styling)
+                img.style = img.getAttribute('style') ? img.removeAttribute('style') : 'height:auto;';
+            // if (!skip_content_styling)
```
Comment (DiscreteConcrete) [inline source/js/about.js:27]: ?

File: source/js/interview.js [hunk 19]
```diff
@@ -185,9 +185,9 @@ $(document).ready(function () {
 
   // Function that's responsible for generating a hash from a string
   function hashCode(string) {
```
Comment (DiscreteConcrete) [inline source/js/interview.js:187]: there are many different types of hashes for different purposes, what are the properties / purposes of this hash function?

File: source/js/modules/Modal.js [hunk 20]
```diff
@@ -108,10 +109,11 @@ class ModalHandler {
 
   /**
      * `Show the modal-box`
+     * @param {Number} time The time for the modal box to slide down
      */
   show() {
-    if (this.sliding) $(this.view.container.parentElement).slideDown(400);
-    else $(this.view.container.parentElement).fadeIn(400);
+    // if (this.sliding) $(this.view.container.parentElement).animate({ height: '100vh' }, time);
```
Comment (DiscreteConcrete) [inline source/js/modules/Modal.js:115]: ?

**Review by DiscreteConcrete — CHANGES_REQUESTED** [review]

File: Extensions/AboutUsParser.js (hunk 2 above)
Comment (AdrianJakubcik) [reply Extensions/AboutUsParser.js:112]: Sorry, an old habit from other languages.

File: Extensions/AboutUsParser.js [hunk 21]
```diff
@@ -0,0 +1,267 @@
...
+        return;
+    }
+}
+
+/** Creates a Json file with our data
+ * 
+ * @param {Array} array Array of paragraph objects
+ */
+function saveAsJson(array) {
+    fs.writeFileSync(C.dPath + C.jsonFile, JSON.stringify(array, null, 4), {flag: 'w+', mode: 0666});
```
[the remark this answers is not in the document]
Comment (AdrianJakubcik) [reply Extensions/AboutUsParser.js:178]: Which codes? What do you mean?

[the remark this answers is not in the document]
Comment (AdrianJakubcik) [reply Extensions/AboutUsParser.js:178]: I'm not sure what you either mean by that or I don't see any usage of having it returned to the caller since I don't do anything with later when it's executed.

File: Extensions/AboutUsParser.js [hunk 22]
```diff
@@ -0,0 +1,267 @@
...
+        }
+    } catch (error) {
+        console.error(error)
+        return;
+    }
+}
+
+/** Creates a Json file with our data
+ * 
+ * @param {Array} array Array of paragraph objects
```
[the remark this answers is not in the document]
Comment (AdrianJakubcik) [reply Extensions/AboutUsParser.js:175]: The array should contain only objects of the type Paragraph, it's a class.

File: Extensions/AboutUsParser.js (hunk 7 above)
Comment (AdrianJakubcik) [reply Extensions/AboutUsParser.js:220]: Depends on the scope they are used in.

File: build.js [hunk 23]
```diff
@@ -460,7 +460,9 @@ if (!success)
 	return;
 // second, we prepare all components
 const enComponents = makeComponentDictionary(getDirname() + '/components/{*,en/*}.html')
+console.log(enComponents);
```
[the remark this answers is not in the document]
Comment (AdrianJakubcik) [reply build.js:463]: It was just a debug message for me it should be commented out now.

File: components/navigation.html (hunk 9 above)
Comment (AdrianJakubcik) [reply components/navigation.html:11]: Because I'm testing out at the moment whether the event `onkeyup` works on mobile devices too. I think I had a problem with the searcher on my phone but not sure.

File: source/css/aboutus.css (hunk 12 above)
Comment (AdrianJakubcik) [reply source/css/aboutus.css:114]: Testing out what looks better whether `height: auto;` or `height: max-content;`, we'll see over time so I'll just leave it there for now.

File: source/js/interview.js (hunk 19 above)
Comment (AdrianJakubcik) [reply source/js/interview.js:187]: to generate a hash code that can be used to identify certain things

Comment (AdrianJakubcik) [reply source/js/interview.js:187]: the sole purpose and function of this function is to just generate a hash from given string

File: Extensions/AboutUsParser.js [hunk 24]
```diff
@@ -0,0 +1,267 @@
...
+        this.long = object.text ? object.text.long : object.long;
+        this.language = object.language;
+        this.folder = object.folder;
+        this.photo = object.photo ? object.photo : null;
+    }
+
+    getDefaultHtmlCode() {
+            return `<div class="card about">
+        <div class="card-img">
+            ${this.photo ? `<img src="{fill_parents}${this.photo}" class="card-img-top" alt="Picture Not Available!">` : ''}
```
[the remark this answers is not in the document]
Comment (AdrianJakubcik) [reply Extensions/AboutUsParser.js:47]: I use it just technically to inform the user that there was supposed to be a picture but it failed to load. No use for it except that.
